Modules/_randommodule.c

```diff
@@ -275,7 +283,8 @@ random_seed(RandomObject *self, PyObject *arg)
     if (PyLong_Check(arg)) {
         /* Calling int.__abs__() prevents calling arg.__abs__(), which might
            return an invalid value. See issue #31478. */
-        n = PyLong_Type.tp_as_number->nb_absolute(arg);
+        n = PyObject_CallFunctionObjArgs(_randomstate_global->Long___abs__, arg,
```

Comment:
This looks... ugh. It also *looks* like it could potentially change behaviour... and potentially be a lot slower than calling the slot function directly. Perhaps we need a new _Py_ conveniency function for this?

Reply:
The fast way to make this call is to use `_PyObject_Vectorcall`. Technically it is not in the public API yet, but `PyObject_Vectorcall` will be in 3.9

Reply:
I've switched this to doing a PyLong_CheckExact for the common case w/ a dispatch to PyNumber_Absolute if we match.  And in the case that we don't we'll fallback to the vector call.